src/action-sheets/index.js

```diff
@@ -214,6 +215,49 @@ const muteStream = async ({ auth, streamId, subscriptions }) => {
 muteStream.title = 'Mute stream';
 muteStream.errorMessage = 'Failed to mute stream';
 
+const hashReplacements: Map<string, string> = new Map([
+  ['%', '.'],
+  ['(', '.28'],
+  [')', '.29'],
+  ['.', '.2E'],
+]);
+
+// we trying to create uri like web app.
+// see https://github.com/zulip/zulip/blob/HEAD/static/js/hash_util.js
+const encodeHashComponent = (str: string) =>
+  encodeURIComponent(str).replace(/[%().]/g, matched => hashReplacements.get(matched) ?? '');
+
+const getStreamDetail = (streamId, streams) => {
+  const stream = streams.get(streamId);
+  invariant(stream !== undefined, 'Stream with provided streamId not found.');
+
+  // The name part of the URL doesn't really matter, so we try to
+  // make it pretty.
+  const streamName = stream.name.replace(' ', '-');
+
+  const streamDetail = `${streamId}-${streamName}`;
+  return encodeHashComponent(streamDetail);
+};
+
+const getStreamUri = (streamId, streams): string =>
+  `#narrow/stream/${getStreamDetail(streamId, streams)}`;
+
+const getFullUrl = (domainName, streamId, streams) => {
+  const restUri = getStreamUri(streamId, streams);
+  const url = `${domainName}/${restUri}`;
+  return url;
+};
+
+const copyLinkToStream = async ({ _, streamId, streams, auth }) => {
+  const realm = auth.realm;
+  const serverSettings: ApiResponseServerSettings = await api.getServerSettings(realm);
+  const uri = getFullUrl(serverSettings.realm_uri, streamId, streams);
```

Comment:
Let's use "url" instead of "uri". There's some interesting background on this choice at https://chat.zulip.org/#narrow/stream/92-learning/topic/URLs/near/794826.